lib/s3.js

```diff
@@ -0,0 +1,113 @@
+'use strict';
+
+const assert = require('assert');
+const awsSDK = require('aws-sdk');
+const logMeGently = require('@alpha-lambda/log-me-gently');
+const uuid = require('uuid');
+
+const AwsDriverError = require('./aws-driver-error');
+
+class S3Driver {
+  constructor({ AWS, client, conf = {}, level }) {
+    if (client) {
+      this._client = client;
+    } else {
+      const clientOptions = Object.assign({
+        apiVersion: '2006-03-01',
+        signatureVersion: 'v4'
+      }, conf.client);
+
+      const aws = AWS || awsSDK;
+      this._client = new aws.S3(clientOptions);
+    }
+
+    this._conf = conf;
+    this._logger = logMeGently({ level });
+  }
+
+  get client() {
+    return this._client;
+  }
+
+  get conf() {
+    return this.conf;
+  }
+
+  getSignedUrl(context, { bucket, operation, key, urlTTL }) {
+    assert(bucket, 'missing bucket');
+    assert(operation, 'missing operation');
+    assert(key, 'missing key');
+    assert(urlTTL !== undefined, 'missing urlTTL');
+
+    const getParams = {
+      Bucket: bucket,
+      Expires: urlTTL,
+      Key: key
+    };
+
+    this._logger.log(context, { getParams, operation }, 'generating signed URL');
+    return new Promise((resolve, reject) => {
+      this._client.getSignedUrl(operation, getParams, (err, res) => {
```

Comment:
just sanity check, this is because it has no .promise?

Reply:
yes